### PR TITLE
EEG_data should publish output to subscribers

### PR DESCRIPTION
**Pull request by gregbci**
[B4K-227](https://bci4kids.atlassian.net/browse/B4K-227)

I removed the LSL outlet dependency by injecting a "messenger" object.  To keep things simple, the messenger is just a delegate, taking over responsibility for sending an event message, ex: "marker received".  If we want to implement a true observer pattern (ex: publish / subscribe), we can make a messenger object that does that.  Either way, EEG_data shouldn't have to care.  At this point, I don't think pub/sub is needed.

All the "offline test" examples and existing unit tests don't need a messenger, so I made the default "None".  If there isn't a messenger available, EEG_data won't try sending messages.

I added an LslMessenger for all the "unity backend" examples.  They create it and pass to EEG_data.

I added another test for the "ping" message in test_eeg_data.py.  This demonstrates using a mock Messenger object for testing, in this case, counting pings after running a few loops.

### CODE REVIEW

File: tests/test_xdf_sources.py
Comment (brianjohannes) [inline tests/test_xdf_sources.py:1]: Just checking to make sure that taking Logger out of this file, I see it is left in some of the other files.

File: tests/test_lsl_sources.py
Comment (brianjohannes) [inline tests/test_lsl_sources.py:1]: Same here, just checking to make sure that taking Logger out of this file, I see it is left in some of the other files.

File: tests/data/test_data.xdf
Comment (brianjohannes) [inline tests/data/test_data.xdf:1]: Could we call this xdf_test_data.xdf or test_xdf_sources_data.xdf? I wasn't sure what it was for until I searched. Also, why keep it here and not just use one from examples/data? OR should we copy the examples/data xdfs that are used into this folder as well?

**Review by brianjohannes — APPROVED** [review]
Looks good to me! Only things of note are that you removed some Logger lines and wanted to make sure that was intentional. Also, curious if having a data file in tests is ideal if we can leverage from examples. I'm fine with keeping it, but think the xdf should be renamed.

File: tests/test_xdf_sources.py
Comment (gregbci) [reply tests/test_xdf_sources.py:1]: Oh, yeah... Anup had added some logging for the exceptions in the tests.  But these weren't errors, the tests were checking that an exception was raised.  Once I removed that logger wasn't needed.

File: tests/test_lsl_sources.py
Comment (gregbci) [reply tests/test_lsl_sources.py:1]: Same here, after removing the unnecessary logging.  Hmmm, we might add this back... let's wait for Anup's logger changes to be complete first.

File: tests/data/test_data.xdf
Comment (gregbci) [reply tests/data/test_data.xdf:1]: Yep, I can rename, "test" is a bit too generic :slightly_smiling_face:   

I copied the data file over to tests/ for stability - if test data changes, the tests can break in hard to follow ways.  We don't want that.

The examples, on the other hand, should be free to change as needed without worrying about breaking a test.

So... yeah, I'm thinking example data is separate from test data.

Comment (gregbci) [reply tests/data/test_data.xdf:1]: Oh... and yes... in my next PR, I have copied other xdfs into tests/data (p300, ssvep, mi)

Sorry, this should have been a separate change (wagging finger at myself)
